util/kb.c

```diff
@@ -393,7 +393,10 @@ redis_new (kb_t *kb, const char *kb_path)
 
   kbr = g_malloc0 (sizeof (struct kb_redis) + strlen (kb_path) + 1);
   kbr->kb.kb_ops = &KBRedisOperations;
-  strncpy (kbr->path, kb_path, strlen (kb_path));
+  #pragma GCC diagnostic push
+  #pragma GCC diagnostic ignored "-Wstringop-overflow"
+    strncpy (kbr->path, kb_path, strlen (kb_path) - 1);
+  #pragma GCC diagnostic pop
```

Comment:
```suggestion
    strncpy (kbr->path, kb_path, strlen (kb_path) - 1);
```

Reply:
This change will not work:
[ 97%] Built target gvm_base_shared
In file included from /usr/include/string.h:638,
                 from /usr/include/glib-2.0/glib/gtestutils.h:30,
                 from /usr/include/glib-2.0/glib.h:82,
                 from /builddir/build/BUILD/gvm-libs-20.8.1/util/../base/nvti.h:30,
                 from /builddir/build/BUILD/gvm-libs-20.8.1/util/kb.h:28,
                 from /builddir/build/BUILD/gvm-libs-20.8.1/util/kb.c:27:
In function 'strncpy',
    inlined from 'redis_new' at /builddir/build/BUILD/gvm-libs-20.8.1/util/kb.c:396:3:
/usr/include/bits/string3.h:120:10: error: '__builtin_strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  120 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/gvm-libs-20.8.1/util/kb.c: In function 'redis_new':
/builddir/build/BUILD/gvm-libs-20.8.1/util/kb.c:396:32: note: length computed here
  396 |   strncpy (kbr->path, kb_path, strlen (kb_path) - 1);
      |                                ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors